.github/workflows/golangci.yml

```diff
@@ -0,0 +1,28 @@
+name: golangci-lint
```

Comment:
Can we not run this as part of precommit?

Reply:
We can, the advantage of this approach is that the github action will automatically create notes on the relevant diffs

Reply:
OK, that makes it worthwhile